perf/tuple.cc

```diff
@@ -147,6 +203,39 @@ class MpData<FORMAT_BASIC> {
 	char *data_end;
 };
 
+template<>
+class MpData<FORMAT_SPARSE> {
+public:
+	const char *begin() const { return data; }
+	const char *end() const { return data_end; }
+	MpData()
+	{
+		size_t field_count =
+			TupleFormat<FORMAT_SPARSE>::instance().field_count();
+		std::vector<bool> is_non_null(field_count);
+		for (size_t i = 0; i < FIELD_COUNT_NON_NULL; i++)
+			is_non_null[rand() % field_count] = true;
```

Comment:
AFAICS this doesn't guarantee that the number of non-null fields will be exactly `FIELD_COUNT_NON_NULL`. Anyway, I don't understand why you need this temporary array, i.e. why you don't call `rand` from within the next loop.

Reply:
It's OK for the test.